src/high-level/util.lisp

```diff
@@ -95,3 +95,19 @@
 
 (defun normalize-type (type)
   (upgraded-array-element-type type))
+
+(defmacro define-extensible-function ((fun-name fun-name-backend &optional (backend :lisp)) lambda-list &body options)
+  "This macro mimics DEFGENERIC, except instead it makes FUN-NAME a backend function whose BACKEND implementation (default :LISP) is a generic function named FUN-NAME-BACKEND."
+  (let* ((doc-option (assoc ':documentation options))
+         (doc-string (if (null doc-option)
+                         nil
+                         (second doc-option))))
+    `(progn
+       ;; Only define backend functions for :LISP. We don't want to
```

Comment:
Would it be reasonable to keep a list of defined backend functions and only generate them if it does not already exist instead of relying on there always being a lisp implementation?

Reply:
yes; done